pkg/deps/config/suppliers.go

```diff
@@ -507,3 +513,75 @@ func newSupplyTxClientsFn(
 
 	return depinject.Configs(deps, depinject.Supply(txClient)), nil
 }
+
+// NewSupplyKeyValueCacheFn returns a function which constructs a KeyValueCache of type T.
+// It take a list of cache options that can be used to configure the cache.
+func NewSupplyKeyValueCacheFn[T any](opts ...querycache.CacheOption[cache.KeyValueCache[T]]) SupplierFn {
+	return func(
+		ctx context.Context,
+		deps depinject.Config,
+		cmd *cobra.Command,
+	) (depinject.Config, error) {
+		// Check if query caching is enabled
+		queryCachingEnabled, err := cmd.Flags().GetBool(FlagQueryCaching)
+		if err != nil {
+			return nil, err
+		}
+
+		// Use a NoOpKeyValueCache if query caching is disabled
```

Comment:
NIT: Do you mind adding a log before & after this if structure just stating "Query Caching is enabled/disabled".

If it ever becomes an issue in debugging, we'll have this data point right away.

Reply:
Added in https://github.com/pokt-network/poktroll/blob/b1ef0add07118296d3fc84faa6a4043b52ba3bad/pkg/relayer/cmd/cmd.go#L150-L154